aten/src/ATen/native/quantized/cpu/qnnpack/src/qnnpack/requantization.h

```diff
@@ -278,31 +260,13 @@ pytorch_qnnp_compute_scalar_avgpool_quantization_params(
   /* Compute requantization parameters */
   assert(scale >= 0x1.0p-32f);
   assert(scale < 256.0f);
-  const uint32_t scale_bits = fp32_to_bits(scale);
-
-  /* Multiplier is in [0x00800000, 0x00FFFFFF] range */
-  const int32_t multiplier =
-      ((int32_t)scale_bits & INT32_C(0x007FFFFF)) | INT32_C(0x00800000);
-  assert(multiplier >= INT32_C(0x00800000));
-  assert(multiplier <= INT32_C(0x00FFFFFF));
-
-  /* Shift is in [16, 55] range */
-  const int32_t shift = 127 + 23 - (scale_bits >> 23);
-  assert(shift >= 16);
-  assert(shift < 64);
 
   union pytorch_qnnp_avgpool_quantization_params params;
-  const uint32_t right_shift = (uint32_t)shift;
-  const int64_t rounding = INT64_C(1) << (right_shift - 1);
   params.scalar.bias = bias;
-  params.scalar.rounding = rounding;
-  params.scalar.multiplier = multiplier;
-  params.scalar.right_shift = right_shift;
-  params.scalar.output_min_less_zero_point =
-      (int32_t)(uint32_t)output_min - (int32_t)(uint32_t)output_zero_point;
-  params.scalar.output_max_less_zero_point =
-      (int32_t)(uint32_t)output_max - (int32_t)(uint32_t)output_zero_point;
+  params.scalar.scale = scale;
```

Comment:
Not sure, but do we have assertions anywhere in the code that require the scale to be <1 ? Or did we only have them for conv/linear?

Reply:
Yes only for conv/linear. Add and avgpool had different constraints. Not < 1. But for conv/linear I missed them in the integrated code. So I need to fix it.